### PR TITLE
Diagrammes partners

### PR DESCRIPTION
**Pull request by AnaisVlr**
Revoir mes diagrammes (séquences + classes)

### CODE REVIEW

**Comment by unixyy** [comment]
for the class diagram, don't we need to add the Partner class itself ?

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 1]
```diff
@@ -0,0 +1,98 @@
+//CREATE PARTNER
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: `'` for comments

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+participant PartnerPage
+participant PartnerController
+participant PartnerFacade
+participant AbstractFactory
+participant PartnerDAO
+participant DataBase
+
+activate Admin
+
+PartnerPage -> Admin: displayPartnerList()
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: not sure it's needed, as the use case start when the admin is _on_ the partners page

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+Admin -> PartnerPage++: enterName
+Admin -> PartnerPage: enterDescription
+Admin -> PartnerPage: enterWebsite
+Admin -> PartnerPage: clickOnCreatePartner
+PartnerPage -> PartnerController++: onClickButtonCreatePartner()
+PartnerController -> PartnerFacade++: createPartner()
+PartnerFacade -> AbstractFactory++: createPartnerDAO()
+
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML:23]:
```plantuml
PartnerController -> PartnerFacade++: createPartner(): Partner|Null
PartnerFacade -> AbstractFactory++: createPartnerDAO()
return partnerDAO
PartnerFacade -> PartnerDAO: createPartner(): Partner
```

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    PartnerController --> PartnerPage: createSucessful()
+    PartnerPage --> Admin--: display(partnerListWindow)
+else Partner already exist
+    PartnerController -> PartnerPage: createFailed()
+    PartnerPage --> Admin--: display(invalid)
+end
+
+@enduml
+
+//UPDATE PARTNER
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: same remarks as above

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    PartnerController --> PartnerPage: updateSucessful()
+    PartnerPage --> Admin--: display(partnerListWindow)
+else Partner doesn't exist
+    PartnerController -> PartnerPage: updateFailed()
+    PartnerPage --> Admin--: display(invalid)
+end
+
+@enduml
+
+//DELETE PARTNER
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: same here

File: doc/reports/final/partners/partners_classDiagram.plantUML [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+database "Postgres\nDatabase" as Postgres {
+
+}
+
+PartnerPage -> PartnerController
+PartnerController -> PartnerFacade
+PartnerFacade -> AbstractFactory
+AbstractFactory -> PartnerDAO
+Postgres -> AbstractFactory
+PartnerFacade -> PartnerDAO
```
Comment (iusildra) [inline doc/reports/final/partners/partners_classDiagram.plantUML]: You should add the labels and quantities. This gives you a better view:
```plantuml
PartnerPage       -u-> PartnerController
PartnerController --> PartnerFacade
PartnerFacade     -u-> AbstractFactory
AbstractFactory   -u-> PartnerDAO
Postgres          --> AbstractFactory
PartnerFacade     --> PartnerDAO
```

**Review by iusildra — COMMENTED** [review]
Nice work ! Just a few changes suggested

**Comment by iusildra** [comment]
> for the class diagram, don't we need to add the Partner class itself ?

I don't think so, we already have the model in some other diagrams. We just need to show the processes

File: doc/reports/final/partners/partners_classDiagram.plantUML [hunk 7]
```diff
@@ -1,41 +1,74 @@
...
+Interface PartnerDAO {
+  + {abstract} createPartner()
+  + {abstract} updatePartner()
+  + {abstract} deletePartner()
 }
 
-database "Postgres\nDatabase" as Postgres {
 
+class PostgresPartnerDAO{
+  + getPostgresPartnerDAO() : PartnerDAO
```
Comment (iusildra) [inline doc/reports/final/partners/partners_classDiagram.plantUML]: same here
```suggestion
  + {static} getPostgresPartnerDAO() : PartnerDAO
```

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 8]
```diff
@@ -13,32 +11,33 @@ participant DataBase
...
-@startuml
+@startuml <Update Partner>
 skinparam responseMessageBelowArrow true
 
 actor Admin
-participant PartnerPage
+participant View.Main
 participant PartnerController
 participant PartnerFacade
 participant AbstractFactory
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: the separator should be in the diagram
```plantuml
end

=== Update partners ==

activate Admin
...
```
But this diagram format is also good

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 9]
```diff
@@ -47,32 +46,33 @@ participant DataBase
...
-Admin -> PartnerPage: clickOnUpdatePartner
-PartnerPage -> PartnerController++: onClickButtonUpdatePartner()
+Admin -> View.Main++: enterInformations
+Admin -> View.Main: onClickButtonUpdatePartner()
+View.Main -> PartnerController++: updatePartner()
 PartnerController -> PartnerFacade++: updatePartner()
-PartnerFacade -> AbstractFactory++: updatePartnerDAO()
+PartnerFacade -> AbstractFactory++: getPartnerDAO()
+return PartnerDAO
+PartnerFacade -> PartnerDAO++: updatePartnerDAO()
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: a resistant
```suggestion
PartnerFacade -> PartnerDAO++: updatePartner()
```

File: doc/reports/final/partners/partners_sequenceDiagram.plantUML [hunk 10]
```diff
@@ -81,18 +81,21 @@ participant DataBase
...
-PartnerPage -> Admin: displayPartnerList
-Admin -> PartnerPage: clickOnDeletePartner
-PartnerPage -> PartnerController++: onClickButtonDeletePartner()
+Admin -> View.Main++: onClickButtonDeletePartner()
+View.Main -> PartnerController++: deletePartner()
 PartnerController -> PartnerFacade++: deletePartner()
-PartnerFacade -> AbstractFactory++: deletePartnerDAO()
+PartnerFacade -> AbstractFactory++: getPartnerDAO()
+return PartnerDAO
+PartnerFacade -> PartnerDAO++: deletePartnerDAO()
```
Comment (iusildra) [inline doc/reports/final/partners/partners_sequenceDiagram.plantUML]: another one
```suggestion
PartnerFacade -> PartnerDAO++: deletePartner()
```

File: doc/reports/final/partners/partners_classDiagram.plantUML [hunk 11]
```diff
@@ -1,41 +1,74 @@
...
 class PartnerController {
+  + createPartner()
+  + deletePartner()
+  + updatePartner()
+  + createSucessful()
+  + updateSucessful()
+  + deleteSucessful()
+  + createFailed()
+  + updateFailed()
+  + deleteFailed()
```
Comment (iusildra) [inline doc/reports/final/partners/partners_classDiagram.plantUML]: imo, it's better to have a single method with a boolean
```suggestion
  + createSucessful(boolean)
  + updateSucessful(boolean)
  + deleteSucessful(boolean)

File: doc/reports/final/partners/partners_classDiagram.plantUML [hunk 12]
```diff
@@ -1,41 +1,74 @@
...
+interface PartnerAbstractFactory {
+  + {abstract} getPartnerDAO() : PartnerDAO
+
 }
 
-abstract PartnerDAO {
-  + getPartnerById()
+
+class PostgresPartnerFactory {
+  + getPostgresPartnerFactory(): PostgresPartnerFactory
```
Comment (iusildra) [inline doc/reports/final/partners/partners_classDiagram.plantUML]:
```suggestion
  + {static} getPostgresPartnerFactory(): PostgresPartnerFactory
```

**Review by iusildra — COMMENTED** [review]
Some suggestions, but last ones for me !

**Review by AnaisVlr — COMMENTED** [review]
ok

**Pull request merged**